packages/teams-js/test/public/app.spec.ts

```diff
@@ -11,585 +11,1213 @@ import {
-      const initPromise = app.initialize();
+        const initMessage = utils.findMessageByFunc('initialize');
+        utils.respondToMessage(
+          initMessage,
+          FrameContexts.content,
+          HostClientType.web,
+          '{"apiVersion":1, "supports":{"mail":{}}}',
+        );
+        await initPromise;
 
-      const initMessage = utils.findMessageByFunc('initialize');
-      utils.respondToMessage(initMessage, FrameContexts.content, HostClientType.web, '', '1.6.0');
-      await initPromise;
+        expect(runtime).not.toEqual(teamsRuntimeConfig);
+        expect(runtime).toEqual({ apiVersion: 1, supports: { mail: {} } });
+      });
 
-      expect(runtime).toEqual(teamsRuntimeConfig);
-    });
+      it('app.initialize should assign clientSupportedSDKVersion correctly when a proper runtime config is given', async () => {
+        const initPromise = app.initialize();
+
+        const initMessage = utils.findMessageByFunc('initialize');
+        utils.respondToMessage(
+          initMessage,
+          FrameContexts.content,
+          HostClientType.web,
+          '{"apiVersion":1, "supports":{"mail":{}}}',
+          '1.0.0',
+        );
+        await initPromise;
+
+        expect(runtime).toEqual({ apiVersion: 1, supports: { mail: {} } });
+        expect(GlobalVars.clientSupportedSDKVersion).toBe('1.0.0');
+      });
+
+      it('app.initialize should initialize with clientSupportedSDKVersion and runtimeConfig arguments flipped', async () => {
+        const initPromise = app.initialize();
+
+        const initMessage = utils.findMessageByFunc('initialize');
+        utils.respondToMessage(
+          initMessage,
+          FrameContexts.content,
+          HostClientType.web,
+          '1.0.0',
+          '{"apiVersion":1, "supports":{"mail":{}}}',
+        );
+        await initPromise;
+
+        expect(runtime).toEqual({ apiVersion: 1, supports: { mail: {} } });
+        expect(GlobalVars.clientSupportedSDKVersion).toBe('1.0.0');
+      });
 
-    it('app.initialize should use teams runtime config if a JSON parsing error is thrown by a given runtime config', async () => {
-      const initPromise = app.initialize();
+      it('app.initialize should initialize with teams config when an invalid runtimeConfig is given, with arguments flipped', async () => {
+        const initPromise = app.initialize();
 
-      const initMessage = utils.findMessageByFunc('initialize');
-      utils.respondToMessage(initMessage, FrameContexts.content, HostClientType.web, 'nonJSONStr', '1.6.0');
-      await initPromise;
+        const initMessage = utils.findMessageByFunc('initialize');
+        utils.respondToMessage(initMessage, FrameContexts.content, HostClientType.web, '1.6.0', 'nonJSONStr');
+        await initPromise;
 
-      expect(runtime).toEqual(teamsRuntimeConfig);
+        expect(runtime).toEqual(teamsRuntimeConfig);
+      });
     });
```

Comment:
These are great additions! Unless I'm missing something, I think there are some more test cases that should be added to validate `app.initialize`:

- validating GlobalVars.hostClientType is set correctly
- validating the various capabilities `.initialize` functions are called (e.g., `authentication.initialize()`, `menus.initialize()`, etc.)
- validating GlobalVars.additionalValidOrigins is set correctly